File: src/1768_merge_strings_alternately/MergeStringsAlternately.java
public class MergeStringsAlternately {
    public String mergeAlternately(String word1, String word2) {
        int n1 = word1.length(), n2 = word2.length();
        StringBuilder res = new StringBuilder();
        for (int i = 0; i < Math.max(n1, n2); ++i) {
            if (i < n1)
                res.append(word1.charAt(i));
            if (i < n2)
                res.append(word2.charAt(i));
        }
        return res.toString();
    }

    // Test scaffolding
    public static void main(String[] args) {
        MergeStringsAlternately sol = new MergeStringsAlternately();
        System.out.println(sol.mergeAlternately("abc", "pqr")); // apbqcr
        System.out.println(sol.mergeAlternately("ab", "pqrs")); // apbqrs
        System.out.println(sol.mergeAlternately("abcd", "pq")); // apbqcd
        System.out.println(sol.mergeAlternately("a", "z")); // az
        System.out.println(sol.mergeAlternately("abcde", "")); // abcde
    }
}

File: src/1768_merge_strings_alternately/merge_strings_alternately.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
   public:
    string mergeAlternately(string word1, string word2) {
        int n1 = word1.size(), n2 = word2.size();
        string res;
        for (int i = 0; i < max(n1, n2); ++i) {
            if (i < n1) res += word1[i];
            if (i < n2) res += word2[i];
        }
        return res;
    }
};

// Test scaffolding
int main() {
    Solution sol;
    cout << sol.mergeAlternately("abc", "pqr") << endl;  // apbqcr
    cout << sol.mergeAlternately("ab", "pqrs") << endl;  // apbqrs
    cout << sol.mergeAlternately("abcd", "pq") << endl;  // apbqcd
    cout << sol.mergeAlternately("a", "z") << endl;      // az
    cout << sol.mergeAlternately("abcde", "") << endl;   // abcde
    return 0;
}
